File: shell/WordCount.scala
:sh mkdir -p target
:sh rm -rf target/wordcount.txt
{
  sc.textFile("../data/input.txt")
        .flatMap(_.split("[^\\w]"))
        .map(_.toLowerCase().trim())
        .filter(!_.isEmpty())
        .map(word => (word, 1))
        .reduceByKey(_ + _)
        .map(_.swap)
        .sortByKey()
        .saveAsTextFile("target/wordcount.txt")
}

File: shell/WordCount.py
import os
os.system("mkdir -p target")
os.system("rm -rf target/wordcount.txt")

wordcount = sc.textFile('../data/input.txt') \
    .flatMap(lambda s: s.split(' ')) \
    .filter(lambda w:len(w) > 0) \
    .map(lambda w:(w.lower(),1)) \
    .reduceByKey(lambda a,b: a + b) \
    .map(lambda (word,count):(count, word)) \
    .sortByKey(False).map(lambda (count,word):(word, count)) 
wordcount.saveAsTextFile("target/wordcount.txt")
